src/templates/_select.twig

```diff
@@ -1,11 +1,121 @@
+{% set now = "now"|date("U") %}
+
+<div style="margin-bottom: 5px; margin-top: 5px;">
+    {% if selected_only_class is defined and selected_only_class %}
+        <span style="float: left;">
+            <input type="checkbox" class="{{ selected_only_class }}_{{ now }}" value=""><span style="margin-left: 5px;">{{ 'Show selected only'|t }}</span>
+        </span>
+    {% endif %}
+
+    {% if clear_selected_class is defined and clear_selected_class %}
+        <span style="float: right;">
+            <button class="{{ clear_selected_class }}_{{ now }}">{{ 'Clear selected'|t }}</button>
+        </span>
+    {% endif %}
+</div>
+
 {% if name is defined and name %}
-	<input type="hidden" name="{{ name }}" value="">
+    <input type="hidden" name="{{ name }}" value="">
 {% endif %}
 <div class="multiselect">
-	<select name="{{ name }}[]" multiple>
-		<option value="">{{ "None"|t }}</option>
-		{% for option in products %}
-			<option value="{{ option.productId }}"{% if option.productId in value %} selected{% endif %}>{{ option.label }}</option>
-		{% endfor %}
-	</select>
-</div>
\ No newline at end of file
+    <select style="width: 100%; height: 20rem;" class="{{ name }}_{{ now }}" name="{{ name }}[]" multiple>
+        <option value="">{{ "None"|t }}</option>
+        {% for option in products %}
+            <option value="{{ option.productId }}"{% if option.productId in value %} selected{% endif %}>{{ option.label }} ({{ option.sku }})</option>
+        {% endfor %}
+    </select>
+</div>
+
+{% if filter_class is defined and filter_class %}
+    <input style="width: 99.5%;" placeholder="{{ 'Start typing to filter products ...'|t }}" type="text" class="{{ filter_class }}_{{ now }}" value="">
+{% endif %}
+
+
+<script>
+(() => {
+    const add_date_class = (element) => {
+        const right_now = Date.now();
+        if (!/_extra_date_/.test(element.classList)) {
+            element.classList.add(`_extra_date_${right_now}`);
+        }
+    };
+
+    const attach_events = () => {
+        let commerce_input = document.querySelector(
```

Comment:
Also scope all query selectors to maybe an root element. Something like:
```JavaScript
const $root = document.querySelector('.craft-shopify-plugin');
...
let commerce_input = $root.querySelecot('input[class="{{ filter_class }}_{{ now }}"]');
```

Reply:
Well, they need access to the twig variables and need to be dynamically created for times when you have multiple versions of the field on one entry, otherwise the JS will not work. I am fine with moving the CSS out.

Reply:
Good callout.

Reply:
This is fixed:
https://github.com/nmaier95/craft-shopify-product-fetcher/pull/9/commits/85c5edee1ed84e44017030e19fb7884588a257bf